### DISCUSSION TITLE
ch2/ch2-03

### DISCUSSION THREAD

**Discussion by giscus** [gopl-zh/gopl-zh.github.com#207] [General]
# ch2/ch2-03

<The Go Programming Language>中文版

https://golang-china.github.io/gopl-zh/ch2/ch2-03.html

**Comment by dahezhiquan** [comment]
不错

**Comment by PanandSu** [comment]
Good

**Comment by 8vd8** [comment]
打卡，看这篇有点累

**Comment by windqiu** [comment]
打卡，一个个关卡一章章学

**Comment by siqi47-a** [comment]
打卡，终于理解了*x和&x的区别。

**Comment by xiaoxu-365** [comment]
let`s go

**Comment by zhusl-2021** [comment]
“由于new只是一个预定义的函数，它并不是一个关键字，因此我们可以将new名字重新定义为别的类型。”
这样用不是自找麻烦嘛，何必呢😂

**Comment by wlwqq** [comment]
每次循环的开始会创建临时变量t

这句话有点歧义，其实在for循环只会创建并初始化一次，在每一轮循环时会执行t += res

**Reply by Raoxu20030404** [reply]
+1
我是go的初学者，t := 0.0应该在循环开始前只执行一次并且循环结束前不会销毁，才符合初始化语句的含义与程序逻辑吧

**Comment by chaserZH** [comment]
1

**Comment by Raven762L** [comment]
打卡，指针看的有点晕了

**Comment by weiyi-ch** [comment]
生命周期：从每个包级的变量和每个当前运行函数的每一个局部变量开始，通过指针或引用的访问路径遍历，是否可以找到该变量。

**Comment by Chen-05** [comment]
<pre>
func incr(p *int) int {
    *p++ // 非常重要：只是增加p指向的变量的值，并不改变p指针！！！
    return *p
}

v := 1
incr(&v)              // side effect: v is now 2
fmt.Println(incr(&v)) // "3" (and v is 3)

"每次我们对一个变量取地址，或者复制指针，我们都是为原变量创建了新的别名。例如，*p就是变量v的别名。"
我个人觉得这里对别名的认知是错误的，因为指针变量p与变量v本身都会有自己的内存地址，它们是两个独立的空间。而别名的意思更多是两个变量名指向同一个内存地址，两个变量名共用一个内存空间，才叫做别名。
</pre>

**Comment by xqnjkycx** [comment]
nice
